pages/index.tsx

```diff
@@ -44,26 +42,19 @@ export default function HomePage({ packageCount }) {
 }
 
 export const getServerSideProps: GetServerSideProps = async (_context) => {
+  let packageCount;
   try {
-    const response = await fetch(`${API_URL}/api/count/packages/`);
-    //to prevent the case where site crashes due to error response
-    if (!response || response.status!==200) {
-      return {
-        props: {
-          packageCount: "all"
-        }
-      }
-    }
-    const { packageCount } = await response.json();
-    return {
-      props: {
-        packageCount
-      }
-    };
+    const response = await fetch(`${API_URL}/api/packages?limit=${Number.MAX_SAFE_INTEGER}`, {
+      method: 'HEAD'
+    });
+    packageCount = response.headers.get('x-total-count');
   } catch (error) {
-    // if package doesn't exist, return a 404
-    return {
-      notFound: true,
-    };
+    packageCount = null;
```

Comment:
Since you defined the type in the component to be `{ packageCount?: number }` (it's an optional props), you can set a default value to `undefined`, this way the prop will be ignored

Reply:
Could you clarify this? When I set a packageCount to undefined it throws an error